MBBSEmu/Memory/IntPtr16.cs

```diff
@@ -10,13 +10,21 @@ public class IntPtr16 : IEquatable<IntPtr16>
         public ushort Segment
         {
             get => BitConverter.ToUInt16(Data, 2);
-            set => Array.Copy(BitConverter.GetBytes(value), 0, Data, 2, 2);
+            set
+            {
+                Data[2] = (byte)(value & 0xFF);
+                Data[3] = (byte)(value >> 8);
+            }
 
         }
         public ushort Offset
         {
             get => BitConverter.ToUInt16(Data, 0);
-            set => Array.Copy(BitConverter.GetBytes(value), 0, Data, 0, 2);
+            set
+            {
+                Data[0] = (byte)(value & 0xFF);
+                Data[1] = (byte)(value >> 8);
+            }
         }
 
         public readonly byte[] Data = new byte[Size];
```

Comment:
I wonder if this would be even more efficient if it were an array of ushort, so you could read/write a word at a time?

Reply:
I'll benchmark changing the getter and setter for Data so the segment/offset use a backing ushort and Data is built on request from those values.

I suspect we use .Segment and .Offset way more frequently than .Data, which I believe is only used when we're committing the value to memory.